column.go

```diff
@@ -628,15 +628,15 @@ func (c *Column) decodeDataPage(header DataPageHeader, numValues int, repetition
 
 	pageKind := pageType.Kind()
 	if pageKind >= 0 && int(pageKind) < len(pageValuesBufferPool) {
-		vbuf = pageValuesBufferPool[pageKind].get()
+		vbuf = pageValuesBufferPool[pageKind].get(int(pageType.EstimateSize(numValues)))
 		defer vbuf.unref()
 		vbuf.resize(int(pageType.EstimateSize(numValues)))
 		pageValues = vbuf.data
 	}
 	if pageKind == ByteArray {
-		obuf = pageOffsetsBufferPool.get()
+		obuf = pageOffsetsBufferPool.get(4 * (numValues + 1))
 		defer obuf.unref()
-		obuf.resize(4 * (numValues + 1))
+		obuf.resize(4 * (numValues + 1)) // jpe - wut?
```

Comment:
If you intend to keep this comment, do you mind expanding it a bit to carry more context? It's a bit hard to grasp to meaning as-is.

Reply:
apologies, i leave my initials "jpe" as reminders to go back and review code. this shouldn't be here.